services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/UpdateItemOperation.java

```diff
@@ -80,8 +91,12 @@ public UpdateItemRequest generateRequest(TableSchema<T> tableSchema,
             throw new IllegalArgumentException("UpdateItem cannot be executed against a secondary index.");
         }
 
-        Map<String, AttributeValue> itemMap = tableSchema.itemToMap(this.request.item(),
-                                                                    Boolean.TRUE.equals(this.request.ignoreNulls()));
+        T item = request.map(UpdateItemEnhancedRequest::item, TransactUpdateItemEnhancedRequest::item);
+        Boolean ignoreNulls = request.map(r -> Optional.ofNullable(r.ignoreNulls()),
+                                          r -> Optional.ofNullable(r.ignoreNulls()))
```

Comment:
Is it immaterial whether the item itself is left or right when checking for ignorenulls?

Reply:
`generateRequest` is used both on the transact and *NON*-transact path, so it has to be without further refactoring/duplication.

I see the concern that the result of `generateRequest` could be used directly when `request` is actually a `Transact*EnhancedRequest`, but since this is an internal class the chances are low; the only code paths that use the `create()` overloads that take the transact versions of the requests are on `TransactWriteItemsEnhancedRequest`.

Reply:
Yes, we don't care about that so much. I can see though that the number of possible code paths in the library is increasing with every change such as this one and opens up for future edge use cases :-). It's inevitable though.